### PR TITLE
Fix core typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Continuation of #37541 but focused on core (C++).

Top typos are:
* convertable (10)
* transfered (7)
* enpoint (6)
* addtional (5)
* begining (3)
* mutiple (3)
* mesage (3)
* compresed (3)
* overriden (3)
* prefered (3)
* bechmark (3)
* targetted (3)

### CODE REVIEW

File: src/core/lib/experiments/config.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ namespace grpc_core {
 struct ExperimentMetadata {
   const char* name;
   const char* description;
-  const char* additional_constaints;
```
Comment (ctiller) [inline src/core/lib/experiments/config.h:34]: I think this one is going to cause a bunch of grief when we import it internally - can we exclude it from this change and do it separately?

Comment (NathanBaulch) [reply src/core/lib/experiments/config.h]: Sure thing.

File: src/core/server/server.h [hunk 2]
```diff
@@ -531,7 +531,7 @@ struct grpc_server_config_fetcher {
   class ConnectionManager
       : public grpc_core::DualRefCounted<ConnectionManager> {
    public:
-    // Ownership of \a args is transfered.
+    // Ownership of \a args is transferred.
```
Comment (markdroth) [inline src/core/server/server.h:534]: This comment can actually be removed entirely.  It's obselete now that we're using a C++ type for the channel args.

Comment (NathanBaulch) [reply src/core/server/server.h]: Done.

**Review by markdroth — APPROVED** [review]

**Comment by markdroth** [comment]
Looks like the sanity check is failing.  I think you need to run `./tools/distrib/clang_format_code.sh`.

The other test failures look unrelated.

**Review by ctiller — APPROVED** [review]

**Pull request closed without merging**